src/app/features/courses/courses.component.ts

```diff
@@ -1,30 +1,57 @@
 import { Component, OnInit } from '@angular/core';
-import { mockedAuthorsList, mockedCoursesList } from '@shared/mocks/mocks';
 import { Course } from '@shared/models/course.model';
-import { mapAuthorIdsToNames } from '@app/utilities/author-utils';
+import { CoursesStoreService } from '@app/services/courses-store.service';
+import { UserStoreService } from '@app/user/services/user-store.service';
+import { Router } from '@angular/router';
+import { ButtonText } from '@shared/models/button.model';
 
 @Component({
   selector: 'app-courses',
   templateUrl: './courses.component.html',
   styleUrls: ['./courses.component.scss'],
 })
 export class CoursesComponent implements OnInit {
-  courses: Course[] = mockedCoursesList;
-  mappedCourses: { course: Course; authorsNames: string[] }[] = [];
-  selectedCourse!: Course;
+  // mappedCourses: { course: Course; authorsNames: string[] }[] = [];
+  courses!: Course[];
+  selectedCourse!: Course | null;
+  courseForEdit!: Course;
+  isAdmin!: boolean;
+
+  constructor(
+    private coursesStoreService: CoursesStoreService,
+    private userService: UserStoreService,
+    private router: Router
+  ) {}
 
   ngOnInit() {
-    this.mappedCourses = this.courses.map(course => ({
-      course,
-      authorsNames: mapAuthorIdsToNames(course.authors, mockedAuthorsList),
-    }));
+    this.coursesStoreService.getAll();
+    this.coursesStoreService.getAllAuthors();
+    this.coursesStoreService.courses$.subscribe(data => (this.courses = data));
+    this.userService.getUser();
+    this.userService.isAdmin$.subscribe(isAdmin => (this.isAdmin = isAdmin));
   }
 
-  onCourseSelected(course: Course) {
+  onShowCourse(course: Course) {
     this.selectedCourse = course;
+    this.router.navigate(['courses', course.id]);
+  }
+
+  onEditCourse(course: Course) {
+    this.courseForEdit = course;
+    this.router.navigate(['courses/edit', course.id]);
+  }
+
+  onDeleteCourse(course: Course) {
+    this.coursesStoreService.deleteCourse(course.id);
+  }
+
+  onClickAddCourse() {
+    this.router.navigate(['courses/add']);
   }
 
   onSearch(searchInput: string): void {
-    // TODO: add logic to filter courses based on input
+    this.coursesStoreService.filterCourses(searchInput);
```

Comment:
Can you please handle the use-case where search term is empty? Maybe even block the request if there is no valid search text present.
![image](https://github.com/user-attachments/assets/f41acbfa-71f8-4182-9064-3198deca998c)

Reply:
Done!